### DISCUSSION TITLE
Possible bug (remix): backend getAuth returning null but frontend useAuth returning logged in user

### DISCUSSION THREAD

**Discussion by setvik** [clerk/javascript#1885] [Support request - DO NOT OPEN HERE]
Pinging before i dive in further to see if anyone else is seeing this.

I have a Remix 2.0.1 Express installation with Clerk 3.0.6. Clerk works fine locally with the "development" Clerk environment, but when i deploy and use the "production" Clerk environment, I can sign in as a user in the frontend without issue, but the Remix backend does not see the user: getAuth() says the user is signed out.

Anyone have thoughts on what might be causing this?

**Comment by mitchatinvestin** [comment]
I'm observing the exact same thing - as soon as I've moved into production, `getAuth` no longer seems to work. 
So far I'm completely stumped as to why, and the docs don't say anything about it. 

Did you have any luck?

**Comment by mitchatinvestin** [comment]
Ahh right okay so it just simply doesn't work, as I think you noticed over in sO haha.

For anyone else - [this link](https://stackoverflow.com/questions/77376975/self-hosted-next-js-app-with-clerk-dev-auth-session-not-recognized-in-productio) has the solution.

**Comment by JUNIORCO** [comment]
Having the same issue. `const { user } = await getAuth(args);` in my loaders has `user` as undefined, but the hook `const { user } = useUser();` works fine. 

Edit: Ah you're supposed to do `const user = await createClerkClient({ secretKey: process.env.CLERK_SECRET_KEY }).users.getUser(userId);`

**Comment by jabelic** [comment]
I was using @clerk/nextjs (4.25.7) and had a similar issue. Upgrading to version 4.29.3 resolved the bug.

**Comment by jescalan** [comment]
https://github.com/orgs/clerk/discussions/3829
